tests/fixtures/odl.py

```diff
@@ -0,0 +1,262 @@
+import json
+import types
+from typing import Any, Callable, Optional, Tuple
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+
+from api.odl import ODLAPI
+from core.model import (
+    Collection,
+    IntegrationClient,
+    Library,
+    License,
+    LicensePool,
+    Loan,
+    Patron,
+    Representation,
+    Work,
+    get_one_or_create,
+)
+from core.testing import DatabaseTest, MockRequestsResponse
+from core.util.http import HTTP
+from tests.fixtures.api_odl_files import ODLAPIFilesFixture
+from tests.fixtures.database import DatabaseTransactionFixture
+
+
+class MonkeyPatchedODLFixture:
```

Comment:
I don;t see this class being used anywhere except in the return of the monkeypatch fixture below, do we need this class?

Reply:
Checking now. It's possible that fixture method is used by name, but the fixture parameter isn't annotated with a type in tests. I've tried to annotate everywhere, but there may be some I've missed.

Reply:
It's used as a parameter for the `ODLTestFixture` class (which is definitely used in tests).

Reply:
Yes, I saw `odl_test_fixture` uses `monkey_patch_odl`, but the class `MonkeyPatchedODLFixture` itself has no use since it only holds the `self.monkeypatch` attribute, which is never used. Maybe `return monkeypatch` directly from `monkey_patch_odl`

Reply:
Or the patch `ODLAPI` class

Reply:
This is a types-as-documentation thing. I'm trying to get the code to the point where the presence of a value of a given type demonstrates that an operation has taken place. If you can observe a value of type `MonkeyPatchedODLFixture`, then you know "a monkey patching operation has done something with the ODL code". If you just return `monkeypatch`, then all you know is "someone patched something".